prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -142,6 +142,26 @@ data Doc ann =
 instance Semigroup (Doc ann) where
     (<>) = Cat
     sconcat (x :| xs) = hcat (x:xs)
+    stimes n x
+      | n <= 0    = Empty
+      | n == 1    = x
+      | otherwise = case x of
+            Fail            -> Fail
+            Empty           -> Empty
+            Char c          -> Text n' (T.replicate n' (T.singleton c))
+            Text l t        -> Text (n' * l) (T.replicate n' t)
+            Line            -> nx
+            FlatAlt{}       -> nx
+            Cat{}           -> nx
+            Nest{}          -> nx
+            Union{}         -> nx
+            Column{}        -> nx
+            WithPageWidth{} -> nx
+            Nesting{}       -> nx
+            Annotated{}     -> nx
+          where
+            n' = fromIntegral n
+            nx = hcat (replicate n' x)
```

Comment:
Would prefer `let … in case … of` here, looking at this code I started looking for where the `nx` came from